components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -215,6 +218,50 @@ class GleanTest {
         Glean.initialized = true
     }
 
+    @Test
+    fun `queued recorded metrics correctly record during init`() {
+        val counterMetric = CounterMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "counter_metric",
+            sendInPings = listOf("store1")
+        )
+
+        // First set to a pre-init state
+        Glean.initialized = false
+        Dispatchers.API.queueTasks = true
+        Dispatchers.API.testingMode = false
+
+        // This will queue 3 tasks that will add to the metric value once Glean is initialized
+        for (i in 0..2) {
+            counterMetric.add()
+        }
+
+        // Enable testing mode and ensure that no value has been stored yet
+        Glean.enableTestingMode()
+        assertFalse("No value must be stored", counterMetric.testHasValue())
+
+        // Calling resetGlean here will cause Glean to be initialized and should cause the queued
+        // tasks recording metrics to execute
+        resetGlean(clearStores = false)
+
+        // Wait for the queued tasks to run by checking the taskQueue.size  Using a timeout just in
+        // case something hangs up so the test won't stall.
+        runBlocking {
```

Comment:
If we make the queued tasks `testMode` aware, then this becomes much simpler and we won't require timeouts and checking `Dispatchers` internals at all.